pkg/kv/kvserver/protectedts/protectedts.go

```diff
@@ -139,6 +141,30 @@ type Verifier interface {
 	Verify(context.Context, uuid.UUID) error
 }
 
+// Metrics encapsulates the metrics exported by the reconciler.
+type Metrics struct {
+	ReconcilationRuns    *metric.Counter
+	RecordsProcessed     *metric.Counter
+	RecordsRemoved       *metric.Counter
+	ReconciliationErrors *metric.Counter
+}
+
+var _ metric.Struct = (*Metrics)(nil)
+
+// MetricStruct makes Metrics a metric.Struct.
+func (m *Metrics) MetricStruct() {}
+
+// Reconciler provides a mechanism to reconcile protected timestamp records with
+// external state.
+type Reconciler interface {
+	// StartReconciler will start the reconciliation where each record's status is
+	// determined using the record's meta type and meta in conjunction with the
+	// configured StatusFunc.
+	StartReconciler(ctx context.Context, stopper *stop.Stopper) error
+	// Metrics returns the metrics exported by the reconciler.
+	Metrics() *Metrics
```

Comment:
I don't think you need to export this.

Reply:
Since we want to subsume the Reconciler interface in the Provider interface, the initialization of the reconciler needs to happen https://github.com/cockroachdb/cockroach/pull/75688/files#diff-47ad9f8fa32a4e686099e2e89dad4beca96772073e042d746b771bc58ffb9f01R55. So we don't have a handle to the concrete type of the reconciler in server/tenant.go to grab its metrics struct. I removed this method and exported the constructor of the Metrics struct instead. This is then plumbed into ptprovider.New().